lms/db/__init__.py

```diff
@@ -101,6 +103,11 @@ def init(engine, drop=False):
         BASE.metadata.drop_all(engine)
     BASE.metadata.create_all(engine)
 
+    try:
+        engine.execute("select 1 from alembic_version")
+    except sqlalchemy.exc.ProgrammingError:
+        alembic.command.stamp(alembic.config.Config("conf/alembic.ini"), "head")
```

Comment:
As far as I can tell the `alembic` Python module doesn't provide any easy way to find out whether the DB is already stamped. `engine.execute("select 1 from alembic_version")` is much simpler and has been working for us in h

Reply:
The relative path is a bit annoying here. You can fix it with `pkg_resources.resource_filename` but it doesn't really fix the issue as the `script_location` in the config is relative to root as well.

To get this to work correctly, I think you'd need a layout like:

 * `lms/lms/alembic/`
   * `conf.ini`
   * `migrations/`

Then the relative path in the conf would work ok. In this case I'd say it's more effort than it's worth, as we don't do this all the time. But it will explode with different working directories.

But in other places this can make running tests in particular very annoying and should probably be a thing we try to avoid.

Reply:
You can't easily run `make dev` (which is what will run this code) from any other directory anyway and if you do go out of your way to do so, lots of existing stuff in `make dev` already fails :man_shrugging: I don't think it's a concern, no one ever does it